src/Driver/ChromeDriver/Downloader.php

```diff
@@ -147,17 +153,35 @@ private function downloadArchive(Driver $driver): string
     private function getBinaryName(Driver $driver): string
     {
         $operatingSystem = $driver->operatingSystem();
-
-        if ($operatingSystem->equals(OperatingSystem::WINDOWS())) {
-            return self::BINARY_WINDOWS;
-        }
-
-        if ($operatingSystem->equals(OperatingSystem::MACOS())) {
-            return self::BINARY_MAC;
-        }
-
-        if ($operatingSystem->equals(OperatingSystem::LINUX())) {
-            return self::BINARY_LINUX;
+        switch ($this->isJsonVersion($driver)) {
```

Comment:
**Todo:** you need to change the DownloaderTest & VersionResolverTest to support this changes with newer versions. Also good to add new case with the major breaking version.

Reply:
yep i added some tests